skore/src/skore/sklearn/_estimator/metrics_accessor.py

```diff
@@ -444,6 +444,144 @@ def _compute_metric_scores(
+
+        X : array-like of shape (n_samples, n_features), default=None
+            Data on which the prediction was computed. By default, we use the validation
+            set provided when creating the report.
+
+        y : array-like of shape (n_samples,), default=None
+            Target on which the prediction was computed. By default, we use the
+            validation target provided when creating the report.
+
+        Returns
+        -------
+        predict_time : float
+            The time, in seconds.
+
+        Examples
+        --------
+        >>> from sklearn.datasets import load_breast_cancer
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> from sklearn.model_selection import train_test_split
+        >>> from skore import EstimatorReport
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...     *load_breast_cancer(return_X_y=True), random_state=0
+        ... )
+        >>> classifier = LogisticRegression(max_iter=10_000)
+        >>> report = EstimatorReport(
+        ...     classifier,
+        ...     X_train=X_train,
+        ...     y_train=y_train,
+        ...     X_test=X_test,
+        ...     y_test=y_test,
+        ... )
+
+        >>> predict_time = report.metrics.predict_time()  # Predictions on the test data
+        ...
+
+        >>> report.metrics.predict_time() == predict_time  # Predict time is cached
+        True
+
+        >>> report.metrics.predict_time(data_source="train")
+        ...
+
+        >>> report.metrics.predict_time(data_source="X_y", X=X_test, y=y_test)
+        ...
+        """
+        X, y, data_source_hash = self._get_X_y_and_data_source_hash(
+            data_source=data_source, X=X, y=y
+        )
+
+        predict_time_cache_key = (
+            self._parent._hash,
+            data_source,
+            data_source_hash,
+            "predict_time",
+        )
+
+        predict_time = self._parent._cache.get(predict_time_cache_key)
+
+        if predict_time is None:
+            # If the time was not found in the cache, compute predictions
+            _get_cached_response_values(
+                cache=self._parent._cache,
+                estimator_hash=self._parent._hash,
+                estimator=self._parent.estimator_,
+                X=X,
+                response_method="predict",
+                data_source=data_source,
+                data_source_hash=data_source_hash,
+                # In binary classification, will default to
+                # self._parent.estimator_.classes_[1]
+                pos_label=None,
+            )
+
+            predict_time = self._parent._cache.get(predict_time_cache_key)
+
+            if predict_time is None:
+                raise Exception(
+                    "Prediction was computed, but predict time was not found. "
+                    "This error should never be reached."
+                )
```

Comment:
Let's make it an assertion since it is exactly what the assert means rather than a general Exception.